tests/unit_tests/layout/test_ctao_array_layouts.py

```diff
@@ -0,0 +1,214 @@
+        for layout in merged["value"]
+    )
+
+    # Check unchanged layout
+    assert any(
+        layout["name"] == "array2" and sorted(layout["elements"]) == ["tel3", "tel4"]
+        for layout in merged["value"]
+    )
+
+    # Check newly added layout
+    assert any(
+        layout["name"] == "array4" and sorted(layout["elements"]) == ["tel5", "tel6"]
+        for layout in merged["value"]
+    )
+
+
+def test_get_array_element_name():
+    """Test getting array element name from common identifier."""
+    # Test normal case
+    array_element_ids = {
+        "array_elements": [{"id": "T1", "name": "telescope1"}, {"id": "T2", "name": "telescope2"}]
+    }
+
+    assert cta_array_layouts._get_array_element_name("T1", array_element_ids) == "telescope1"
+    assert cta_array_layouts._get_array_element_name("T2", array_element_ids) == "telescope2"
+
+    # Test non-existent id
+    assert cta_array_layouts._get_array_element_name("T3", array_element_ids) is None
+
+    # Test empty array elements
+    empty_elements = {"array_elements": []}
+    assert cta_array_layouts._get_array_element_name("T1", empty_elements) is None
+
+    # Test missing array_elements key
+    empty_dict = {}
+    assert cta_array_layouts._get_array_element_name("T1", empty_dict) is None
+
+
+@patch("simtools.layout.ctao_array_layouts.names")
+def test_get_layouts_per_site(mock_names):
+    """Test getting array layouts per site."""
+    # Mock site determination function
+    mock_names.get_site_from_array_element_name.side_effect = (
+        lambda x: "north" if "N" in x else "south"
+    )
+
+    # Test data
+    site = "north"
+    sub_arrays = {
+        "subarrays": [
+            {"name": "array1", "array_element_ids": ["T1", "T2"]},
+            {"name": "array2", "array_element_ids": ["T3", "T4"]},
+        ]
+    }
+    array_element_ids = {
+        "array_elements": [
+            {"id": "T1", "name": "N_tel1"},
+            {"id": "T2", "name": "N_tel2"},
+            {"id": "T3", "name": "S_tel1"},
+            {"id": "T4", "name": "S_tel2"},
+        ]
+    }
+
+    # Call function
+    layouts = cta_array_layouts._get_layouts_per_site(site, sub_arrays, array_element_ids)
+
+    # Assert results
+    assert len(layouts) == 1
+    assert layouts[0]["name"] == "array1"
+    assert layouts[0]["elements"] == ["N_tel1", "N_tel2"]
+
+    # Test empty subarrays
+    layouts = cta_array_layouts._get_layouts_per_site(site, {"subarrays": []}, array_element_ids)
+    assert len(layouts) == 0
+
+    # Test missing keys
+    layouts = cta_array_layouts._get_layouts_per_site(site, {}, array_element_ids)
+    assert len(layouts) == 0
+
+    # Test array with no matching elements
+    site = "south"
+    layouts = cta_array_layouts._get_layouts_per_site(
+        site,
+        {"subarrays": [{"name": "array1", "array_element_ids": ["T1", "T2"]}]},
+        array_element_ids,
+    )
+    assert len(layouts) == 0
+
+
+def test_retrieve_array_layouts_from_url():
```

Comment:
Would it make sense to actually test the retrival from the common identifiers repository and not only mock it?

Reply:
Unit tests should whenever possible run without external dependencies like database connections or file retrieval. This is why we want to mock here. The integration test is testing that we can retrieve the file from the gitlab.